components/server/ee/src/gitlab/gitlab-app-support.ts

```diff
@@ -61,4 +61,18 @@ export class GitLabAppSupport {
         result.unshift(...ownersRepos);
         return result;
     }
+
+    protected async getAccountAvatarUrl(anyProject: any, providerHost: string): Promise<string> {
+        if (anyProject.owner?.avatar_url) {
```

Comment:
**thought**: Alternatively, we could introduce a custom fallback image using GitLab's logo for the groups that we cannot fetch their custom generated avatar, see [relevant comment](https://github.com/gitpod-io/gitpod/issues/5330#issuecomment-1067954451):

> GitLab auto-generates an avatar based on the group name and using a colored background in combination with the first letter of the group name.

Reply:
Not known. Would you know how I can become part of such a group with less permissions, in order to reproduce this?

Alternatively, I can add debug logs to this PR, and ask you to load that page again. 😊

Reply:
Also, I agree that the fallback image would be nice, but [I couldn't find how to generate it](https://github.com/gitpod-io/gitpod/pull/8824#issuecomment-1067956035):

> Note: When there is no avatar, it would be nice to use the same default avatar as GitLab (i.e. a big letter with a pastel background). However, I couldn't find how to generate this, so for now I left the avatar URL empty when there is no avatar.

Would you know a potential way to find out? 👀

Reply:
@jankeromnes Last time I remember contributing to this[[1](https://gitlab.com/gitlab-org/gitlab-foss/-/merge_requests/17271)], GitLab was using a custom logic to create the icon, see [relevant code](https://gitlab.com/gitlab-org/gitlab/-/blob/b2a22b6e85200ce55ab09b5c765043441b086c96/app/helpers/avatars_helper.rb#L151-161). Maybe this is why this is not available through their API?

Reply:
Ah, thank you so much @gtsiolis for linking the relevant code ❤️ I couldn't find it, and now I've used it to implement a similar defafult-avatar-generator for our GitLab avatar fetching code.